src/ophyd_async/tango/core/_tango_transport.py

```diff
@@ -272,14 +284,9 @@ def subscribe_callback(self, callback: Callback | None):
 
         self._callback = callback
         if self.support_events:
-            """add user callback to CHANGE event subscription"""
-            if not self._eid:
-                self._eid = self._proxy.subscribe_event(
-                    self._name,
-                    EventType.CHANGE_EVENT,
-                    self._event_processor,
-                    green_mode=False,
-                )
+            asyncio.run_coroutine_threadsafe(
+                self._subscribe_to_event(), asyncio.get_running_loop()
+            )
```

Comment:
I was a bit confused by this, as I've never seen this called from inside the event loop. Would this be clearer?
```suggestion
            # We are in a sync function, so have to run the async subscribe_to_event
            # in the background
            asyncio.create_task(self._subscribe_to_event())
```

Reply:
I was using the run_coroutine_threadsafe version while I was debugging. Since we should be in the event loop here, create_task makes more sense. I've made the change as suggested.